ios/Exponent/Kernel/AppLoader/EXAppLoaderExpoUpdates.m

```diff
@@ -207,7 +207,11 @@ - (void)appLoaderTask:(EXUpdatesAppLoaderTask *)appLoaderTask didStartLoadingUpd
 
   _remoteUpdateStatus = kEXAppLoaderRemoteUpdateStatusDownloading;
   [self _setShouldShowRemoteUpdateStatus:update.manifest];
-  [self _setOptimisticManifest:[self _processManifest:update.manifest]];
+  EXManifestsManifest *processedManifest = [self _processManifest:update.manifest];
+  if (processedManifest == nil) {
+    return;
```

Comment:
Seems like we need to call the delegate or else this code will hang:
```
    if (self.delegate) {
      [self.delegate appLoader:self didFailWithError:_error];
    }
```

Reply:
delegate is invoked inside of `_processManifest:update` since the NSException -> NSError logic would've had to been copied around a bunch otherwise. But I agree, not the best code.